Code/Framework/AzCore/AzCore/DOM/DomUtils.h

```diff
@@ -150,25 +202,44 @@ namespace AZ::Dom::Utils
         using WrapperType = DomValueWrapperType<T>;
         auto ExtractOpaqueValue = [&value]() -> AZStd::optional<WrapperType>
         {
-            if (!value.IsOpaqueValue())
+            if constexpr (AZStd::is_pointer_v<WrapperType>)
             {
-                return {};
+                void* valuePointer = TryMarshalValueToPointer(value);
+                if (valuePointer != nullptr)
+                {
+                    return reinterpret_cast<WrapperType>(valuePointer);
+                }
             }
-            const AZStd::any& opaqueValue = value.GetOpaqueValue();
-            if (!opaqueValue.is<WrapperType>())
+            if constexpr (AZ::Internal::HasAZTypeInfoIntrusive<T>::value)
```

Comment:
Why is only Intrusive Type Info being checked here?

This should check if the type has external type info as well via `AZ::HasTypeInfo`

Reply:
I couldn't find public API that did this when I wrote it... and I actually still can't find `AZ::HasTypeInfo` or something similar. Would you mind pointing me at the API? I'm more than happy to switch this out.

Reply:
Ah that's what you meant, my apologies! `AZ::Internal::HasAZTypeInfo` will `static_assert` if type info isn't present, making it not suitable for this check. I'll add a clarifying comment.

Reply:
What do you mean by static assert?
`Internal::HasAZTypeInfo` is used if a type has TypeInfo so it doesn't static assert